src/getindex.jl

```diff
@@ -0,0 +1,143 @@
+const Indexer = AbstractVector{<:Integer}
+
+Base.IndexStyle(::LinearMap) = IndexCartesian()
+# required in Base.to_indices for [:]-indexing
+Base.eachindex(::IndexLinear, A::LinearMap) = Base.OneTo(length(A))
+Base.lastindex(A::LinearMap) = last(eachindex(IndexLinear(), A))
+Base.firstindex(A::LinearMap) = first(eachindex(IndexLinear(), A))
+
+function Base.checkbounds(A::LinearMap, i, j)
+    Base.checkbounds_indices(Bool, axes(A), (i, j)) || throw(BoundsError(A, (i, j)))
+    nothing
+end
+# Linear indexing is explicitly allowed when there is only one (non-cartesian) index
+function Base.checkbounds(A::LinearMap, i)
+    Base.checkindex(Bool, Base.OneTo(length(A)), i) || throw(BoundsError(A, i))
+    nothing
+end
+# checkbounds in indexing via CartesianIndex
+Base.checkbounds(A::LinearMap, i::Union{CartesianIndex{2}, AbstractVecOrMat{CartesianIndex{2}}}) =
+    Base.checkbounds_indices(Bool, axes(A), (i,)) || throw(BoundsError(A, i))
+Base.checkbounds(A::LinearMap, I::AbstractMatrix{Bool}) =
+    axes(A) == axes(I) || throw(BoundsError(A, I))
+
+# main entry point
+function Base.getindex(A::LinearMap, I...)
+    # TODO: introduce some sort of switch?
+    @boundscheck checkbounds(A, I...)
+    _getindex(A, Base.to_indices(A, I)...)
+end
+# quick pass forward
+Base.@propagate_inbounds Base.getindex(A::ScaledMap, I...) = A.λ .* A.lmap[I...]
+Base.@propagate_inbounds Base.getindex(A::WrappedMap, I...) = A.lmap[I...]
+Base.@propagate_inbounds Base.getindex(A::WrappedMap, i::Integer) = A.lmap[i]
+Base.@propagate_inbounds Base.getindex(A::WrappedMap, i::Integer, j::Integer) = A.lmap[i,j]
+
+########################
+# linear indexing
+########################
+function _getindex(A::LinearMap, i::Integer)
+    i1, i2 = Base._ind2sub(axes(A), i)
+    return _getindex(A, i1, i2)
+end
+_getindex(A::LinearMap, I::Indexer) = [_getindex(A, i) for i in I]
+_getindex(A::LinearMap, ::Base.Slice) = vec(Matrix(A))
```

Comment:
This has been a while back, and I almost forgot any details here, but the docstring is instructive:

```julia
Slice(indices)

  Represent an AbstractUnitRange of indices as a vector of the indices themselves, with special handling to signal they represent a complete slice of a dimension (:).

  Upon calling to_indices, Colons are converted to Slice objects to represent the indices over which the Colon spans. Slice objects are themselves unit ranges with the same indices
  as those they wrap. This means that indexing into Slice objects with an integer always returns that exact integer, and they iterate over all the wrapped indices, even supporting
  offset indices.
```

Since `_getindex` is "later in the pipeline", at this point we already handle the output of `to_indices`, which is of `Slice` type:

```julia
julia> Base.to_indices(LinearMap(ones(3,3)), (:,))
(Base.Slice(Base.OneTo(9)),)
```

So, the use of `Slice` is not meant to be in support of offset axes, it's really just so we can have a single

```julia
function Base.getindex(A::LinearMap, I...)
    # TODO: introduce some sort of switch?
    @boundscheck checkbounds(A, I...)
    _getindex(A, Base.to_indices(A, I)...)
end
```

and then deal with the output of `to_indices`.